aws_iot: Update documentation according to AWS certification feedback (#9706)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/aws_iot.rst

> To connect to AWS IoT, the following steps must be completed:

Reviewer: ```suggestion
To connect to AWS IoT, complete the following steps:
```

1. :ref:`setup_aws_and_permissions`
#. :ref:`creating_a_thing_in_AWS_IoT`
#. :ref:`flash_certi_device`
#. :ref:`Configuring the application <configuring>`

See `AWS IoT Developer Guide`_ for general information about the Amazon Web Services IoT service.

.. _setup_aws_and_permissions:

> Setup your AWS account and Permissions
> **************************************

Reviewer: ```suggestion
Set up your AWS account and permissions
***************************************
```

> To connect to AWS IoT you need to setup an AWS account with the approriate permissions.

Reviewer 2: ```suggestion
To connect to AWS IoT you need to setup an AWS account with the appropriate permissions.
```

> To connect to AWS IoT you need to setup an AWS account with the approriate permissions.

Reviewer: ```suggestion
To connect to AWS IoT, you need to set up an AWS account with the appropriate permissions.
```

> This can be done by completing the steps documented in `Set up your AWS account`_.

Reviewer: ```suggestion
Complete the steps documented in `Set up your AWS account`_.
```

.. note::
   If you are using a longer device ID that is either set by the option :kconfig:option:`CONFIG_AWS_IOT_CLIENT_ID_STATIC` or passed in during initialization, it might be required to increase the value of the option :kconfig:option:`CONFIG_AWS_IOT_CLIENT_ID_MAX_LEN` for proper initialization of the library.

Usage
*****

The :ref:`aws_iot` sample showcases the use of this library and can be used to verify a connection to AWS IoT.

> To configure and run the sample, complete the :ref:`_setup_awsiot` and :ref:`aws_iot_sample_building_and_running` steps.

Reviewer: ```suggestion
To configure and run the sample, complete the steps described in :ref:`setup_awsiot` and :ref:`aws_iot_sample_building_and_running`.
```

If this option is set, the ``host_name`` and ``host_name_len`` must be set in the :c:struct:`aws_iot_config` structure before it is passed into the :c:func:`aws_iot_init` function.
The length of your AWS host name (:kconfig:option:`CONFIG_AWS_IOT_BROKER_HOST_NAME`) must be shorter than the default value of :kconfig:option:`CONFIG_AWS_IOT_BROKER_HOST_NAME_MAX_LEN`, for proper initialization of the library.

Testing and debugging
=====================

> If you have issues with the library and/or sample, refer to the following section :ref:`gs_testing`.

Reviewer: ```suggestion
If you have issues with the library or sample, refer to :ref:`gs_testing`.
```
